Lib/ufo2ft/featureWriters/baseFeatureWriter.py

```diff
@@ -2,8 +2,26 @@
 from collections import OrderedDict
 from types import SimpleNamespace
 
+from ufo2ft.errors import InvalidFeaturesData
 from ufo2ft.featureWriters import ast
 
+INSERT_FEATURE_MARKER = r"\s*# Automatic Code"
+
+
+def _collectInsertMarkers(feaFile, insertFeatureMarker):
+    """
+    Returns a dictionary of tuples (block, comment) keyed by feature tag
+    """
+    insertComments = dict()
+    for parent, comment in ast.findCommentPattern(feaFile, insertFeatureMarker):
+        if parent is None:
+            raise InvalidFeaturesData(
+                "Invalid insert marker '%s' outside of feature block" % str(comment)
+            )
+        elif parent.name not in insertComments.keys():
```

Comment:
I think before getting `name` attribute you should assert that parent is an instance of feature block statement, because findCommentPattern searches everywhere there are statements, not just inside feature blocks

Reply:
Done.
I updated findCommentPattern to handle nested blocks and made collectInsertMarkers a static function.